test-e2e/panels/structures.test.ts

```diff
@@ -7,36 +7,119 @@ test('should draw structure and display it with MF', async ({ page }) => {
   // Open the "Structures" panel.
   await nmrium.clickPanel('Structures');
 
-  // The SVG container should not be rendered when there are no molecules.
-  await expect(nmrium.page.locator('.mol-svg-container ')).toBeHidden();
-  // Click on the "Add Molecule" button.
-  await nmrium.page.click('data-test-id=panel-structures-button-add >> nth=0');
-
-  // Select the "aromatic ring" tool.
-  await nmrium.page.click('canvas >> nth=0', {
-    position: {
-      x: 35,
-      y: 200,
-    },
+  await test.step('empty panel', async () => {
+    // The SVG container should not be rendered when there are no molecules.
+    await expect(nmrium.page.locator('.mol-svg-container ')).toBeHidden();
   });
-  // Draw the aromatic ring.
-  await nmrium.page.click('canvas >> nth=1', {
-    position: {
-      x: 50,
-      y: 50,
-    },
+  await test.step('Add aromatic ring molecule', async () => {
+    // Click on the "Add Molecule" button.
+    await nmrium.page.click(
+      'data-test-id=panel-structures-button-add >> nth=0',
```

Comment:
```suggestion
      'data-test-id=panel-structures-button-add',
```

Avoid using `nth=0` if the locator is guaranteed to resolve to only one element.
It's also good to remove the `nth` selector so that if because of a bug there are more elements, the selector will throw an error.

Reply:
@targos 
it is not guaranteed because the structure panel is reused in the prediction panel you should take care in case the prediction panel is opened

Reply:
But in this testcase the prediction panel is not expected to be there so we prefer there is an error if suddenly there is a second structure editor.